Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift

```diff
@@ -0,0 +1,77 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2022-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+
+public class DashboardSettingsViewModel: ObservableObject {
+    // MARK: - Inputs & Outputs
+    @Published public var showGrades: Bool
+    @Published public var colorOverlay: Bool
+
+    // MARK: - Outputs
+    @Published public private(set) var layout: DashboardLayout
+    public let isGradesSwitchVisible: Bool
+    public let isColorOverlaySwitchVisible: Bool
+    public let popoverSize: CGSize
+
+    // MARK: - Inputs
+    public let setGridLayout = PassthroughSubject<Void, Never>()
+    public let setListLayout = PassthroughSubject<Void, Never>()
```

Comment:
We could use a typed subject like `PassthroughSubject<DashboardLayout, Never>()` so we don't need an extra property and mapping.

Reply:
I don't even know why I did this way, nice catch!